crates/types/src/config/invocation.rs

```diff
@@ -21,37 +21,21 @@ use serde::{Deserialize, Serialize};
 pub struct InvocationOptions {
     /// # Default journal retention
     ///
-    /// Default journal retention for all invocations. A value of `0` or `None` means no retention by default.
+    /// Default journal retention for all invocations. A value of `0` means no retention by default.
     ///
     /// In production setups, it is advisable to disable default journal retention,
     /// and configure journal retention per service using the respective SDK APIs.
-    #[serde(
-        with = "serde_with::As::<Option<restate_serde_util::DurationString>>",
-        skip_serializing_if = "Option::is_none",
-        default
-    )]
-    #[cfg_attr(
-        feature = "schemars",
-        schemars(with = "Option<restate_serde_util::DurationString>")
-    )]
-    pub default_journal_retention: Option<Duration>,
+    #[serde(skip_serializing_if = "FriendlyDuration::is_zero", default)]
+    pub default_journal_retention: FriendlyDuration,
```

Comment:
Generally speaking, I personally find a good idea that the data structure itself uses `std::time::Duration` type + `serde(with)`, rather than a wrapper that might be refactored over and over.